lib/diyepw/diyepw/analyze_noaa_isd_lite_files.py

```diff
@@ -0,0 +1,140 @@
+import pandas as _pd
+import os as _os
+from typing import Iterable as _Iterable
+
+def analyze_noaa_isd_lite_files(
+        files: _Iterable,
+        *,
+        max_missing_rows: int,
+        max_consecutive_missing_rows: int,
+        compression:str='infer'
+):
+    """
+    Performs an analysis of a set of NOAA ISD Lite files, determining which of them are suitable
+    for conversion into AMY EPW files.
+
+    :param files: An collection of paths to the NOAA ISD Lite files to be analyzed. The files may be
+       uncompressed, or in any compression format permitted by pandas.read_csv(). If they are
+       compressed, they need to have a corresponding file extension, or the compression parameter
+       must be passed. File names must begin with the WMO index of the weather station at which the
+       observations in the file were recorded.
+    :param max_missing_rows: The maximum number of total missing rows to allow in a file.
+    :param max_consecutive_missing_rows: The maximum number of consecutive missing rows to allow in a file.
+    :param compression: If you pass compressed files that don't end in the typical extension for their
+       compression type (e.g. ".gz" for GZIP, ".zip" for ZIP, etc.) then you must pass this to indicate
+       what the files' compression format is. See the documentation of the `compression` parameter of
+       pandas.read_csv() for more information.
+
+    :return: dict in the form:
+    {
+      "good": [<file_description>, ...],
+      "too_many_total_rows_missing": [<file_description>, ...],
+      "too_many_consecutive_rows_missing": [<file_description>, ...]
+    }
+    ...where <file_description> is itself a dict in the form:
+    {
+      'file': <str> - absolute path to the file,
+      'total_rows_missing': <int> - The number of total rows that are missing data in this file,
+      'max_consec_rows_missing': <int> - The largest number of consecutive rows that are missing data in this file
+    }
+    """
+
+    too_many_missing_rows = []
+    too_many_consecutive_missing_rows = []
+    good_files = []
+
+    # Loop through all .gz files in the input directory - we filter on extension so that things like
+    # README files can be in that path without causing any problems. We filter on files that begin
+    # with 7 because all NOAA ISD Lite files for North America should start with 7, as all WMO indices
+    # in North America do.
+    for file in files:
+
+        # It's nicer to work with absolute paths, especially since we are going to put this path in a
+        # file to share with another script - otherwise that other script needs to know where this
+        # script is located to make sense of the relative paths
+        file = _os.path.abspath(file)
```

Comment:
Use `pkg_resources.resource_filename` to get this data.  It will save issues on different OSs

Reply:
I have a couple of questions about this:

- What OSs have compatibility problems with os.path? My understanding is that cross-OS compatibility is the entire point of the package, so I'm surprised that it would have OS compatibility issues.
- I'm not sure if `pkg_resources.resource_filename` is appropriate here. The files in question here are provided by the caller, and are thus not package resources at all. I may just be misunderstanding the documentation, but my understanding is that `pkg_resources.resource_filename` is intended to be used to get the path to a resource file within a distributed package, not just to generically get a path to some arbitrary file.